src/restruct_poc/model/model.hpp

```diff
@@ -25,39 +25,42 @@ struct Model {
         const Array_6& velocity = Array_6{0., 0., 0., 0., 0., 0.},
         const Array_6& acceleration = Array_6{0., 0., 0., 0., 0., 0.}
     ) {
-        this->nodes.push_back(Node(nodes.size(), position, displacement, velocity, acceleration));
+        this->nodes.push_back(
+            Node(static_cast<int>(nodes.size()), position, displacement, velocity, acceleration)
+        );
         return this->nodes.back();
     }
 
     /// Adds a rigid constraint to the model and returns the constraint.
     Constraint AddRigidConstraint(const Node& node1, const Node& node2) {
-        this->constraints.push_back(
-            Constraint(ConstraintType::kRigid, this->constraints.size(), node1, node2)
-        );
+        this->constraints.push_back(Constraint(
+            ConstraintType::kRigid, static_cast<int>(this->constraints.size()), node1, node2
```

Comment:
We currently encode the Node ID of a target node for a constraint without a target node as -1.  I use this encoding when deciding how to populate the constraint gradient matrix.  I should be able to do this logic in another way, but I wanted to put it off since it will involve more than cosmetic changes to that kernel